docs/mathematics/factorial-of-number.mdx

```diff
@@ -0,0 +1,174 @@
+---
+title: Factorial of a number
+slug: /mathematics/factorial-of-number
+---
+
+## Introduction
+
+Factorial of a number `n` is defined as the product of all positive integers less than or equal to `n`. Factorial function is defined for all non-negative integers. The value of `0!` is `1`.
+
+## Explanation
+
+For a non-negative integer say `n`, factorial of `n` is defined as follows:
+
+```
+Factorial(n) = n * (n - 1) * (n - 2) * . . . . . 3  * 2 * 1
+```
+
+It is also defined as:
+
+```
+Factorial(n) = n * Factorial(n - 1)
+```
+
+Let us have a few examples:
+
+```
+Factorial(5) = 5 * 4 * 3 * 2 * 1 = 120
+Factorial(6) = 6 * 5 * 4 * 3 * 2 * 1 = 720 = 6 * Factorial(5)
+```
+
+## Algorithm
+
+### Recursive
+
+For a given number `n`,
+
+ - If `n` == `1`, return `1`.
+ - Else recursively return `n` * `factorial(n - 1)`
+
+```
+factorial(n):
+    if(n == 1)
+        return 1
+    else
+        return n * factorial(n - 1)
+```
+
+### Iterative
+
+For a given number `n`,
+
+ - Declare a variable factorial and initialize it to `1`.
+ - For `num` starting from `1` to `n`.
+ - Compute `factorial` = `factorial` * `num`
+ - Add the end of the for loop return `factorial`.
+
+```
+factorial = 1
+for i = 1 to n
+  factorial = factorial * i
+return factorial
+```
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+//The length of vector should be >= length of the resultant factorial
+#include<bits/stdc++.h>
+using namespace std;
+
+/*The function takes a number and muiltiplies the number in the school's old fashioned
```

Comment:
There are a few spelling mistakes here like `multiply` is spelled wrong. Kindly get them checked, also please cross heck your whole document once again.